draft: Signal & slot ListBreakpointsHandler

pvmm opened the pull request:
A signal & slot version of `ListBreakpointsHandler`. Advantages:
* `ListBreakpointsHandler` totally detached from `DebuggerForm` (may serve as example for all other commands);
* easier to invoke ` ListBreakpointsHandler` from other widgets (easier to extend widgets it in the future);

Also, `ListBreakpointsHandler` class declaration moved from **DebuggerForm.cpp** to **DebuggerForm.h**, because **moc_\*** files cannot be generated from **\*.cpp** files. And **moc_\*** files need to include `ListBreakpointsHandler` meta-stuff black magic since I put a signal inside `ListBreakpointsHandler` to use those sweet `QObject` powers. This patch includes #86.

pvmm commented:
Also, this patch was tested.

MBilderbeek reviewed (commented):
This "missing connections'  commit is unrelated to the other and belongs to another PR. But I already made these changes in the meantime.

I don't see anything regarding signals and slots in the ListBreakpointsHandler commit... what am I missing? It just looks like a preparation so far...

pvmm commented:
As I said in the comment, this patch includes #86, but it should be ignored (and git would have ignored it) if you have applied #86. But you fixed it yourself. Great. The relevant part of this PR is in the other commit: [ListBreakpointsHandler using signals and slots](https://github.com/openMSX/debugger/pull/87/commits/a1f7024e52ce750223b502c883d604edff342863)

> I don't see anything regarding signals and slots in the ListBreakpointsHandler commit... what am I missing? It just looks like a preparation so far...

What do you mean? I replaced this entire function:
```  
	void replyOk(const QString& message) override
	{
		if (merge) {
			QString bps = form.session.breakpoints().mergeBreakpoints(message);
			if (!bps.isEmpty()) {
				form.comm.sendCommand(new SimpleCommand(bps));
				form.comm.sendCommand(new ListBreakPointsHandler(form, false));
			} else {
				form.disasmView->update();
				form.session.sessionModified();
				form.updateWindowTitle();
			}
		} else {
			form.session.breakpoints().setBreakpoints(message);
			form.disasmView->update();
			form.session.sessionModified();
			form.updateWindowTitle();
		}
		delete this;
	}
```
by this:
```
void ListBreakpointsHandler::replyOk(const QString& message)
{
	emit replyReceived(message, merge);
	delete this;
}
```
and I moved the previous code into function `DebuggerForm::processBreakpoints(const QString& message, bool merge)`

pvmm commented:
Actually, this has become obsolete because of the new Command class.

pvmm commented:
I am reopening this, as this code here is necessary for my next PRs.
